Table.union assertion fails on mismatched requiredness in globals

table1:
```
'Table{global:+Struct{},key:[s],row:Struct{s:String,PC1:Float64,PC2:Float64,PC3:Float64,PC4:Float64,PC5:Float64,PC6:Float64,PC7:Float64,PC8:Float64,PC9:Float64,PC10:Float64,PC11:Float64,PC12:Float64,PC13:Float64,PC14:Float64,PC15:Float64,PC16:Float64,PC17:Float64,PC18:Float64,PC19:Float64,PC20:Float64,source:String}}'
```

table2:
```
'Table{global:Struct{},key:[s],row:Struct{s:String,PC1:Float64,PC2:Float64,PC3:Float64,PC4:Float64,PC5:Float64,PC6:Float64,PC7:Float64,PC8:Float64,PC9:Float64,PC10:Float64,PC11:Float64,PC12:Float64,PC13:Float64,PC14:Float64,PC15:Float64,PC16:Float64,PC17:Float64,PC18:Float64,PC19:Float64,PC20:Float64,source:String}}'
```

`table1.union(table2)` fails with:
```
Java stack trace:
java.lang.AssertionError: assertion failed
    at scala.Predef$.assert(Predef.scala:156)
    at is.hail.expr.TableUnion.<init>(Relational.scala:1996)
    at is.hail.table.Table.union(Table.scala:947)
    at is.hail.table.Table.union(Table.scala:945)
```

--- a/src/main/scala/is/hail/expr/Relational.scala
+++ b/src/main/scala/is/hail/expr/Relational.scala
@@ -2394,10 +2394,9 @@ case class TableExplode(child: TableIR, fieldName: String) extends TableIR {
 }
 
 case class TableUnion(children: IndexedSeq[TableIR]) extends TableIR {
-  assert(children.length > 0)
-  assert(children.tail.forall(_.typ == children(0).typ))
-  assert((children(0).typ.key.isEmpty && children.tail.forall(_.typ.key.isEmpty)) ||
-    children.tail.forall(_.typ.key.exists(_ == children(0).typ.key.get)))
+  assert(children.nonEmpty)
+  assert(children.tail.forall(_.typ.rowType == children(0).typ.rowType))
+  assert(children.tail.forall(_.typ.key == children(0).typ.key))
 
   def copy(newChildren: IndexedSeq[BaseIR]): TableUnion = {
     TableUnion(newChildren.map(_.asInstanceOf[TableIR]))

